src/plugins/intel_gpu/src/graph/impls/common/loop.cpp

```diff
@@ -155,12 +320,32 @@ struct loop_impl : typed_primitive_impl<loop> {
                 }
             }
 
-            //TODO: execution_condition is prepared as they are presented in the
-            //      ngraph opset document for loop operation.
-            // However they are not being used yet and only TensorIterator which
-            // has fixed sequence length is being validated.
-            if (!primitive->condition_id.empty()) {
-                execution_condition = loop_node::read_scalar_value(execution_condition_mem, stream);
+            if (!loop_carried_dep.empty())
+                stream.wait_for_events(loop_carried_dep);
```

Comment:
Original intention of "loop_carried_dep" is to set the dependency for the next execution of body network. 
However, waiting for the output every time, it is no longer "loop carried dep",but  just output.
I am wondering this does not cause perf regression for the existing tensor iterator scenario.
Could you trigger benchmark job with "loop" filter  for kernel?

Reply:
I will check it. thanks.